examples/golang/topic-creation/topic-creation.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sns"
+)
+
+func main() {
+	topic := flag.String("t", "", "The name of the topic")
+	attributes := flag.String("a", "", "Topic attributes needed")
+	attributesmap := map[string]*string{}
+	json.Unmarshal([]byte(*attributes), &attributesmap) // convert JSON string to Go map
+	flag.Parse()
+
+	if *topic == "" {
+		fmt.Println("You must supply the name of the topic")
+		fmt.Println("-t TOPIC")
+		return
+	}
+	//Ceph RGW Cluster credentials
+	access_key := "0555b35654ad1656d804"
+	secret_key := "h7GhxuBLTrlhVUyxSPUKUV8r/2EI4ngqJxD7iBdBYLhwluN30JaT3Q=="
+	token_id := ""
+	url := "http://127.0.0.1:8000"
+
+	defaultResolver := endpoints.DefaultResolver()
+	snsCustResolverFn := func(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
+		if service == "sns" {
+			return endpoints.ResolvedEndpoint{
+				URL: url,
+			}, nil
+		}
+
+		return defaultResolver.EndpointFor(service, region, optFns...)
+	}
+
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		Config: aws.Config{
+			Region:           aws.String("default"),
+			Credentials:      credentials.NewStaticCredentials(access_key, secret_key, token_id),
+			S3ForcePathStyle: aws.Bool(true),
+			EndpointResolver: endpoints.ResolverFunc(snsCustResolverFn),
+		},
+	}))
+
+	client := sns.New(sess)
+
+	results, err := client.CreateTopic(&sns.CreateTopicInput{
+		Attributes: attributesmap,
```

Comment:
this does not seem to work.
after you create the topic, you should call (from the "build" directory):
```
bin/radosgw-admin topic get --topic sample-topic-1
```
and verify that you see your attributes in: `push_endpoint` and `push_endpoint_args`

Reply:
After the last commit, I can see the attributes in `push_endpoint` and `push_endpoint_args`
This might have been an issue sorted out with checking for errors during conversion of json string to golang map.

Reply:
Where can I get the documentation for `bin/radosgw-admin`?